### PR TITLE
Update validation for ASM Pluginca feature

### PR DESCRIPTION
**Pull request by deveshdama**
---

Add validation for keyvault field

```
az aks mesh enable -g rg-name -n mc-name
--key-vault-id /subscriptions/854c9ddb-fe9e-4aea-8d58-99ed88282881/resourceGroups/rg-test/providers/Microsoft.KeyVault/vaults/test-vault
--ca-cert-object-name ca-cert-name
--ca-key-object-name ca-key-name
--root-cert-object-name root-cert-name
--cert-chain-object-name cert-chain-name
```

### Related command
`az aks mesh enable `

### General Guidelines

- [ ] Have you run `azdev style <YOUR_EXT>` locally? (`pip install azdev` required)
- [ ] Have you run `python scripts/ci/test_index.py -q` locally?

For new extensions:

- [ ] My extension description/summary conforms to the [Extension Summary Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/extensions/extension_summary_guidelines.md).

### About Extension Publish

There is a pipeline to automatically build, upload and publish extension wheels.  
Once your pull request is merged into main branch, a new pull request will be created to update `src/index.json` automatically.  
You only need to update the version information in file setup.py and historical information in file HISTORY.rst in your PR but do not modify `src/index.json`.

### CODE REVIEW

**Comment by yonzhan** [comment]
Thank you for your contribution! We will review the pull request and get back to you soon.

File: src/aks-preview/azext_aks_preview/tests/latest/test_aks_commands.py [hunk 1]
```diff
@@ -7432,15 +7434,15 @@ def test_aks_azure_service_mesh_with_pluginca(self, resource_group, resource_gro
 
         # enable azure service mesh with pluginca
         update_cmd = 'aks mesh enable --resource-group={resource_group} --name={name} ' \
-                     '--key-vault-id my-akv-id ' \
+                     '--key-vault-id  {akv_resource_id} ' \
```
Comment (FumingZhang) [inline src/aks-preview/azext_aks_preview/tests/latest/test_aks_commands.py:7437]: Queued [live test](https://dev.azure.com/msazure/CloudNativeCompute/_build/results?buildId=79450589&view=results) to validate the change.

Comment (FumingZhang) [reply src/aks-preview/azext_aks_preview/tests/latest/test_aks_commands.py:7437]: test passed

**Review by FumingZhang — APPROVED** [review]
LGTM

**Review by zhoxing-ms — APPROVED** [review]

**Pull request merged**